mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/impl/ClientState.scala

```diff
@@ -397,33 +397,39 @@ import scala.util.{Failure, Success}
       case Failure(_) => UnobtainablePacketId
     }
 
-    Behaviors.receiveMessagePartial {
-      case AcquiredPacketId(packetId) =>
-        data.remote ! ForwardSubscribe(packetId)
-        serverSubscribe(
-          ServerSubscribe(packetId, data.subscribeData, data.packetRouter, data.settings)
-        )
-      case UnobtainablePacketId =>
-        Behaviors.stopped
-    }
+    Behaviors
+      .supervise(Behaviors.receiveMessagePartial[Event] {
+        case AcquiredPacketId(packetId) =>
+          data.remote ! ForwardSubscribe(packetId)
+          serverSubscribe(
+            ServerSubscribe(packetId, data.subscribeData, data.packetRouter, data.settings)
+          )
+        case UnobtainablePacketId =>
+          throw SubscribeFailed
```

Comment:
Why throw, rather than stopped?

Reply:
The actor performing the subscription (the one you're looking at) is being watched by its parent. The parent needs to understand *how* the child terminates i.e. whether it was because of failure or whether it completes normally. Throwing an exception in the subject actor is the only way that an observing actor can discern this.

Reply:
Ah right!  And why is it preferable in this situation to throw rather than send a message back?